### PR TITLE
UCP/STREAM: drop data if EP was closed

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
UCP keeps EPs after closure if they are connected remotely until worker destroy, so such EP is able to receive data and can be returned by ucp_stream_worker_poll.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5838/ for details (Mellanox internal link).

**Comment by evgeny-leksikov** [comment]
@alex-mikheev pls review

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5845/ for details (Mellanox internal link).

File: src/ucp/stream/stream.h [hunk 1]
```diff
@@ -28,7 +28,9 @@ typedef struct {
  */
 enum {
     UCP_EP_STREAM_FLAG_IS_QUEUED = UCS_BIT(0), /* EP is queued in stream list of worker */
-    UCP_EP_STREAM_FLAG_HAS_DATA  = UCS_BIT(1)  /* EP has data in the match_q */
+    UCP_EP_STREAM_FLAG_HAS_DATA  = UCS_BIT(1), /* EP has data in the match_q */
+    UCP_EP_STREAM_FLAG_VALID     = UCS_BIT(2)  /* EP was closed locally and cant
+                                                * do recv */
```
Comment (alex-mikheev) [inline src/ucp/stream/stream.h]: I think that the flag description is wrong here because flag indicates that ep can receive data.

Comment (evgeny-leksikov) [reply src/ucp/stream/stream.h]: 👍 it was inverted during implementation, will fix it.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5850/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
👍

**Pull request merged**